### PR TITLE
added spot market functions to the library

### PR DESCRIPTION
**Pull request by sbower**

### CODE REVIEW

File: lib/cucloud/ec2_utils.rb [hunk 1]
```diff
@@ -253,5 +257,64 @@ def find_ebs_snapshots(options = {})
...
     end
+
+    # Get a recommendation for a spot bid request.  Given an instance type and
+    # OS we will grab data from a period specified, default is from two weeks to now,
+    # and calculate recommendations for the AZs in the current region
+    # @param instance_type [String] Insrance type to get bid for
+    # @param os [String] OS you whish to run, default linux
+    # @param num_hours [Integer] How many hours to look back, default two weeks
+    # @return [Hash] Reccomendations by region, empty if no viable recommendations
+    def best_bid_price(instance_type, os = DEFAULT_OS, num_hours = TWO_WEEKS)
```
Comment (bharanin) [inline lib/cucloud/ec2_utils.rb]: perhaps rename best_spot_bid_price() to be consistent with other spot method

File: spec/utilities_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  describe '#confidence_interval_99' do
+    it 'should calculate a 99% confidence interval' do
+      stats = data.descriptive_statistics
+      confidence_interval = Cucloud::Utilities.confidence_interval_99(
+        stats[:mean],
+        stats[:standard_deviation],
+        stats[:number]
+      )
+
+      expect(confidence_interval.map!(&:round)).to eq([30, 80])
```
Comment (bharanin) [inline spec/utilities_spec.rb:17]: Should we test exact rather than rounded value?

Comment (sbower) [reply spec/utilities_spec.rb:17]: yes... I was comparing to test runs in excel and it was convenient to round it off

**Comment by bharanin** [comment]
Looks great!

**Pull request merged**